pandas/io/parsers.py

```diff
@@ -1423,6 +1424,26 @@ def __init__(self, kwds):
         # keep references to file handles opened by the parser itself
         self.handles = []
 
+    def _confirm_parse_dates_presence(self, columns):
+        """
+        if user has provided names for parse_dates, check if those columns
+        are available.
+        """
+        if isinstance(self.parse_dates, list):
+            cols_needed = self.parse_dates
+        elif isinstance(self.parse_dates, dict):
+            cols_needed = chain(*self.parse_dates.values())
```

Comment:
why do you need the chain here?

Reply:
This is okay.  The reason is that `parse_dates` as a `dict` maps column names to lists or individual columns.  `chain` is used to "flatten" them out into a single iterable.